src/new/sendTransaction.js

```diff
@@ -256,25 +254,32 @@ class SendTransaction extends EventEmitter {
       throw new WalletError(ErrorMessages.TRANSACTION_IS_NULL);
     }
 
-    this.emit('send-tx-start', this.transaction);
-    const txHex = this.transaction.toHex();
-    txApi.pushTx(txHex, false, (response) => {
-      if (response.success) {
-        this.transaction.updateHash();
-        this.emit('send-tx-success', this.transaction);
-        if (this._unmarkAsSelectedTimer !== null) {
-          // After finishing the push_tx we can clearTimeout to unmark
-          clearTimeout(this._unmarkAsSelectedTimer);
-          this._unmarkAsSelectedTimer = null;
+    const promise = new Promise((resolve, reject) => {
```

Comment:
I feel like if we try to use more `async/await` instead of the Promise class directly, we could improve the code readability.

It is a little unfriendly to the reader.

But this is just a code style suggestion.

Reply:
I agree about being more friendly to read, however the way we have our API calls from the lib we expect one parameter as the callback from the promise resolve. This is not good and should be refactored in the future but right now we can't use only async await with it.

For the new APIs (you can see in the wallet service facade) we are using only async/await.